python/tests/unittests/test_observe.py

```diff
@@ -685,7 +682,8 @@ def test_observe_list():
     want_expectation_value = -1.7487948611472093
     assert assert_close(want_expectation_value, sum, tolerance=1e-2)
 
-
+## [SKIP_TEST]
+@pytest.mark.skip(reason="TypeError: setSpinOperator(): incompatible function arguments on 'results = cudaq.observe(circuit, hamiltonianList, angles)'")
```

Comment:
Fix 
```diff
diff --git a/python/cudaq/runtime/observe.py b/python/cudaq/runtime/observe.py
index 70e5e4ae5..67249a928 100644
--- a/python/cudaq/runtime/observe.py
+++ b/python/cudaq/runtime/observe.py
@@ -67,22 +67,30 @@ Returns:
     if noise_model != None:
         cudaq_runtime.set_noise(noise_model)
 
+    # Process spin_operator if its a list
+    localOp = spin_operator
+    localOp = cudaq_runtime.SpinOperator()
+    if isinstance(spin_operator, list):
+        for o in spin_operator:
+            localOp += o
+        localOp -= cudaq_runtime.SpinOperator()
+    else:
+        localOp = spin_operator
+
     results = None
     if __isBroadcast(kernel, *args):
         results = __broadcastObserve(kernel,
-                                     spin_operator,
+                                     localOp,
                                      *args,
                                      shots_count=shots_count)
-    else:
-        localOp = spin_operator
-        localOp = cudaq_runtime.SpinOperator()
-        if isinstance(spin_operator, list):
-            for o in spin_operator:
-                localOp += o
-            localOp -= cudaq_runtime.SpinOperator()
-        else:
-            localOp = spin_operator
 
+        if isinstance(spin_operator, list):
+            results = [[
+                cudaq_runtime.ObserveResult(p.expectation(o), o, p.counts(o))
+                for o in spin_operator
+            ]
+                       for p in results]
+    else:
         ctx = cudaq_runtime.ExecutionContext('observe', shots_count)
         ctx.setSpinOperator(localOp)
         cudaq_runtime.setExecutionContext(ctx)
```

Reply:
Patched in [d369a0f](https://github.com/NVIDIA/cuda-quantum/pull/1077/commits/d369a0f1b9ffe8635c714f96a151c4695f7a4e44)